sklearn/linear_model/ridge.py

```diff
@@ -1074,10 +1072,8 @@ def fit(self, X, y, sample_weight=None):
         Y = self._label_binarizer.fit_transform(y)
         if not self._label_binarizer.y_type_.startswith('multilabel'):
             y = column_or_1d(y, warn=True)
-        cw = compute_class_weight(self.class_weight,
-                                  self.classes_, Y)
         # modify the sample weights with the corresponding class weight
-        sample_weight *= cw[np.searchsorted(self.classes_, y)]
+        sample_weight *= compute_sample_weight(self.class_weight, y)
```

Comment:
I am not totally familiar with `RidgeClassifierCV`, but note here ^ that I am now computing the implied `cw` variable over `y` rather than `Y`, as it is/was done in `RidgeClassifier`. I did some tests and it appears that `compute_class_weight` does not mind if it gets a one-hot-encoded version of y, or the original long version. Thus, I do not think that this change has any effect, but just putting it out there.